vm/address/src/lib.rs

```diff
@@ -307,3 +308,24 @@ fn address_hash(ingest: &[u8]) -> [u8; 20] {
     hash.clone_from_slice(&digest);
     hash
 }
+
+#[cfg(feature = "json")]
+pub mod json {
+    use super::*;
+    use serde::{Deserialize, Deserializer, Serializer};
+
+    pub fn serialize<S>(m: &Address, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&encode(m))
+    }
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Address, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let address_as_string: &str = Deserialize::deserialize(deserializer)?;
```

Comment:
This should probably be a `Cow<'de, str>` type, if the string being deserialized has any escape characters, this could fail when it could otherwise pass (and this doesn't allocate in any other case obviously).